### PR TITLE
ENH gaia masking code, buffer regions, other fixes

### LINKED ISSUE #2: cut edges before writing to disk
Need to cut the slice buffers before writing to disk.

### LINKED ISSUE #5: incorporate star masking code from pizza cutter
The stars will now be in the pizza file.

There is a PR on pizza cutter to remove the masking code, so would need to check out a previous version

### PR DESCRIPTION
**Pull request by esheldon**
closes #5 
closes #2

### CODE REVIEW

**Comment by esheldon** [comment]
this now requires installing pizza cutter since the star bitmask value is defined there

**Comment by beckermr** [comment]
It is in the requirements. Do we need to cut a release there?

**Comment by esheldon** [comment]
ahh I see.  Yes, we need a release then.

**Comment by beckermr** [comment]
Done. 0.3.0 should git conda by the end of the day

**Review by beckermr — COMMENTED** [review]
We should make the gaia stuff optional so we don't have to muck with it for certain tests later. 

We'll have to pass in an opaque reference to the input data through joblib. 

I can do this later if you want.

**Comment by beckermr** [comment]
My English got away from me there. We can pass a dictionary with optional keys, one of which is the Gaia star data.

**Comment by esheldon** [comment]
We are going to need a way to test all this as well

**Comment by beckermr** [comment]
yup - fake meds files here we come!

File: bin/run-metadetect-on-slices [hunk 1]
```diff
@@ -37,7 +37,8 @@ def _process(*,
     with open(config, 'r') as fp:
         _config = yaml.load(fp.read(), Loader=yaml.Loader)
 
-    _preconfig = _config.pop("pizza-cutter-preprocessing", None)
+    # required entry
+    _preconfig = _config.pop("pizza-cutter-preprocessing")
```
Comment (beckermr) [inline bin/run-metadetect-on-slices:41]:
```suggestion
    _preconfig = _config.pop("pizza-cutter-preprocessing", {})
```

File: bin/run-metadetect-on-slices [hunk 2]
```diff
@@ -37,7 +37,8 @@ def _process(*,
     with open(config, 'r') as fp:
         _config = yaml.load(fp.read(), Loader=yaml.Loader)
 
-    _preconfig = _config.pop("pizza-cutter-preprocessing", None)
+    # required entry
```
Comment (beckermr) [inline bin/run-metadetect-on-slices:40]:
```suggestion
```

File: pizza_cutter_metadetect/run_metadetect.py [hunk 3]
```diff
@@ -288,14 +306,16 @@ def run_metadetect(
         A multiband MEDS data structure.
     output_file : str
         The file to which to write the outputs.
+    seed: int
+        Base seed for generating seeds
+    preconfig : dict
+        Proprocessing configuration.  Must contain gaia_star_masks
+        entry
```
Comment (beckermr) [inline pizza_cutter_metadetect/run_metadetect.py:313]:
```suggestion
    preconfig : dict
        Proprocessing configuration.  May contain gaia_star_masks
        entry.
```

File: config_files/metadetect-v2.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+# things interpolated using the spline
+spline_interp_flags: 3155
+
+# replaced with noise
+noise_interp_flags: 908
+
+# pixels will have these flag set in the ormask if they were interpolated
+# plus adding in tapebump and star
+imperfect_flags: 20479
+
```
Comment (beckermr) [inline config_files/metadetect-v2.yaml:61]:
```suggestion
```

**Comment by beckermr** [comment]
OK @esheldon This PR is ready for another round of review!

File: pizza_cutter_metadetect/run_metadetect.py [hunk 5]
```diff
@@ -108,40 +117,83 @@ def _make_output_array(
...
+        The size of the central region.
 
     Returns
     -------
     array with new fields
     """
     add_dt = [
         ('slice_id', 'i8'),
-        ('mcal_step', 'S7'),
+        ('mcal_step', 'U7'),
```
Comment (beckermr) [inline pizza_cutter_metadetect/run_metadetect.py:131]: fun with python3 and unicode!

**Comment by esheldon** [comment]
LGTM

**Pull request merged**